### ISSUE TITLE
Cannot create FITS file from slice

### ISSUE THREAD

**Issue by olebole** [radio-astro-tools/spectral-cube#216]
When I try the following:
```Python
from spectral_cube import SpectralCube

cube = SpectralCube('MUSE_Datacube.fits', hdu='DATA')
cube[1].write('image1.fits') # This works
cube[0].write('image0.fits') # This works
(cube[1]-cube[0]).write('diff.fits') # This crashes
```
I get the following stacktrace:
```
ERROR: AttributeError: 'Slice' object has no 'hdu' member [astropy.units.quantity]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-efdec9a72e49> in <module>()
----> 1 (cube[1]-cube[0]).write('a.fits')

/home/oles/.local/lib/python2.7/site-packages/spectral_cube-0.2.2-py2.7.egg/spectral_cube/spectral_cube.pyc in write(self, filename, format, overwrite)
    124             format = determine_format(filename)
    125         if format == 'fits':
--> 126             self.hdu.writeto(filename, clobber=overwrite)
    127         else:
    128             raise ValueError("Unknown format '{0}' - the only available "

/home/oles/.local/lib/python2.7/site-packages/astropy/units/quantity.pyc in __getattr__(self, attr)
    702                 "'{0}' object has no '{1}' member".format(
    703                     self.__class__.__name__,
--> 704                     attr))
    705 
    706         def get_virtual_unit_attribute():

AttributeError: 'Slice' object has no 'hdu' member
```
```

**Comment by keflavich** [comment]
@olebole thanks for pointing this out.  It looks like the problem is that the difference between two quantities does not appropriately inherit the 'hdu' attribute from either of its members.

The workaround for now:
```
hdu = cube[0].hdu
hdu.data = (cube[1]-cube[0]).value
hdu.writeto('file.fits')
```

**Comment by keflavich** [comment]
Related to #123

**Comment by e-koch** [comment]
Addressed in #265
